pkg/apis/kops/cluster.go

```diff
@@ -296,6 +296,10 @@ type KubeDNSConfig struct {
 	Domain string `json:"domain,omitempty"`
 	// ServerIP is the server ip
 	ServerIP string `json:"serverIP,omitempty"`
+	// ClusterCIDR is the CIDR, from the internal network, where we allocate IPs for services
+	ClusterCIDR string `json:"clusterCIDR,omitempty"`
+	// EnableCoreDNS indicates whether CoreDNS or kube-dns will be the default service discovery.
+	EnableCoreDNS bool `json:"enableCoreDNS,omitempty"`
```

Comment:
So I would prefer `provider` because it lets us start to make this more pluggable.  I would love to get to the point where CoreDNS published a k8s manifest, and we could reference it via that URL.  (And that `CoreDNS` or whatever value we choose here means "use the recommended version of CoreDNS").  The reason is that today if CoreDNS (or kube-dns) publishes a new version, we have to bump kops.  In a way that's correct, because kops clusters should be created the same way, but in a way that's wrong because it is already at the breaking point with the network-provider churn.

So I'd really like a string and we recognize two values: "CoreDNS" and "KubeDNS" today.  And an empty string means "choose based on my k8s version".

I'm open to it being a bool, but it should be a `*bool` because otherwise once we've moved to CoreDNS everyone will be stuck with "enableCoreDNS: true` in their manifests, even when KubeDNS is a distant memory.